### DISCUSSION TITLE
Why is `inital` for stack navigation not false by default anymore?

### DISCUSSION THREAD

**Discussion by pke** [react-navigation/react-navigation#10576] [Q&A]
When navigating to a stack navigator, that has never been mounted, the 5.x default is to set the initial page of the navigator to the newly navigated one instead of keeping the specified `initialRouteName` in the navigator.

Why was this behaviour changed since 4.x and in which scenario does this make any sense?

**Comment by satya164** [comment]
> Why was this behaviour changed since 4.x

Because a lot of people complained about the behavior. There's no good default for all use cases. Currently, it's customizable so you can use it as you want.

> in which scenario does this make any sense

In which scenario does pressing the "back button" to return to the screen you came from not make sense?

**Reply by pimJNGNL** [reply]
@satya164 what if your apps get's opened via a deeplink url -> `myapp://user/transactions` for example (which is not the initialRoute!), which is a nested stack inside a TabNavigator along with 3,4 other stacks. The initial stack would be `Stack1` and this `user` stack is `Stack5`.. now I'll get deep linked into `Stack5` via `Linking.openUrl()` which handles the url. On `goBack()` we're now taken back towards `Stack1` instead of the root/initialRoute of `Stack5`. As external deep links are handled differently, I don't see a usage for the `initial` prop on the internal `navigation`.  Is there a way how to solve this, it worked just fine in V4.. without using the `lazy` prop (renders too much). 🙏

**Reply by satya164** [reply]
https://reactnavigation.org/docs/configuring-links/#rendering-an-initial-route

**Reply by pimJNGNL** [reply]
@satya164 thanks a lot!

**Comment by mauricedoepke** [comment]
@pke you might find some work arounds here: https://github.com/react-navigation/react-navigation/issues/10686
